src/SignalR/server/Core/src/HubOptionsExtensions.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.SignalR.Internal;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public static class HubOptionsExtensions
+    {
+        public static void AddFilter(this HubOptions options, IHubFilter hubFilter)
+        {
+            if (options.HubFilters == null)
+            {
+                options.HubFilters = new List<IHubFilter>();
+            }
+
+            options.HubFilters.Add(hubFilter);
+        }
+
+        public static void AddFilter<T>(this HubOptions options) where T : IHubFilter
+        {
+            if (options.HubFilters == null)
+            {
+                options.HubFilters = new List<IHubFilter>();
+            }
+
+            options.HubFilters.Add(new HubFilterFactory<T>());
+        }
+
+        public static void AddFilter(this HubOptions options, Type filterType)
+        {
+            typeof(HubOptionsExtensions).GetMethods()
```

Comment:
We just need a HubFilterFactory implementation to take Type no?

Reply:
Sure. But if you're using reflection anyway to activate the type, is it so bad to do this here?

Reply:
Yes, this is awful 😄